hwsd/filter.cpp

```diff
@@ -135,7 +135,7 @@ class SessionFilter
 public:
     explicit SessionFilter( const std::string& name ) : name_( name ) {}
 
-    const std::string& name_;
+    const std::string name_;
```

Comment:
In this case it's clearly a bug, since you can construct SessionFilter( "foo" ) where foo is a temporary std::string ref'd above.